src/goto-programs/goto_contractor.cpp

```diff
@@ -104,39 +103,58 @@ void goto_contractort::parse_intervals(irep_container<expr2t> expr)
 void goto_contractort::insert_assume(goto_functionst goto_functions)
 {
   loopst loop;
+  unsigned int last_loc = 0;
   for(auto &function_loop : function_loops)
-    loop = function_loop;
+
+    if(
+      last_loc <
+      function_loop.get_original_loop_head()->location_number)
+    {
+      loop = function_loop;
+      last_loc = loop.get_original_loop_head()->location_number;
+    }
 
   auto loop_exit = loop.get_original_loop_exit();
 
   goto_programt dest(message_handler);
 
   auto goto_function = goto_functions.function_map.find("c:@F@main")->second;
 
-  for(auto const &var : map.var_map)
+  if(map.isemptyVector)
```

Comment:
Can we replace `map.isemptyVector` with a `map.var_map.empty()` and delete `map.isemptyVector`?

Reply:
done